test/extensions/filters/http/header_to_metadata/header_to_metadata_filter_test.cc

```diff
@@ -463,6 +463,27 @@ TEST_F(HeaderToMetadataTest, NoEmptyValues) {
   EXPECT_EQ(Http::FilterHeadersStatus::Continue, filter_->decodeHeaders(headers, false));
 }
 
+/**
+ * Missing case is not executed when header is present.
+ */
+TEST_F(HeaderToMetadataTest, NoMissingWhenHeaderIsPresent) {
+  const std::string config = R"EOF(
+request_rules:
+  - header: x-version
+    on_header_missing:
+      metadata_namespace: envoy.lb
+      key: version
+      value: some_value
+      type: STRING
+)EOF";
+  initializeFilter(config);
+  Http::TestRequestHeaderMapImpl headers{{"x-version", ""}};
```

Comment:
nit: add value to the header (to avoid conflating with the internal handling of headers without a value).

Reply:
Updated the PR. <3